packages/react-native/ReactAndroid/src/main/java/com/facebook/react/runtime/ReactHostImpl.java

```diff
@@ -1000,6 +1000,23 @@ private Task<Void> callAfterGetOrCreateReactInstance(
               return null;
             },
             executor)
+        .continueWithTask(
+          task -> {
+            final BridgelessReactContext reactContext = getOrCreateReactContext();
+
+            reactContext.runOnJSQueueThread(
```

Comment:
This causes this event to escape the async Task setup workflow, and will lead to inconsistencies if we're also trying to teardown.

Additionally, you've added this to `callAfterGetOrCreateReactInstance`, which means the event will be called many multiple times during the lifecycle which is incorrect.

Reply:
Okay that's absolutely terrible 😅 I will think about doing it the other way!